### PR TITLE
[Master] Add examples for `lang.typedesc` langlib functions

### LINKED ISSUE #38982: [Improvement]: Add examples for `lang.typedesc` langlib functions
### Description

$title

### Describe your problem(s)

_No response_

### Describe your solution(s)

_No response_

### Related area

-> Other Area

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #38982

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [x] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by SasinduDilshara — CHANGES_REQUESTED** [review]

File: langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal [hunk 1]
```diff
@@ -40,6 +40,12 @@ public type TypeId readonly & record {|
 |};
 
 # Returns the type-ids induced by a typedesc value.
+# 
+# ```ballerina
+# type MyError distinct error;
+# typedesc:TypeId[]? & readonly typeIds = MyError.typeIds();
```
[the remark this answers is not in the document]
Comment (MaryamZi) [reply langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal:46]: Let's add another example passing true for `primaryOnly`.

File: langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal [hunk 2]
```diff
@@ -40,6 +40,22 @@ public type TypeId readonly & record {|
...
 # Returns the type-ids induced by a typedesc value.
+# 
+# ```ballerina
+# type Error distinct error;
+# type SampleError distinct (Error & error<record {string msg;}>);
+# 
+# typedesc:TypeId[]? & readonly typeIds = SampleError.typeIds();
+# if (typeIds is typedesc:TypeId[]) {
+#   io:println(typeIds[typeIds.length() - 1]["localId"]) ⇒ Error
+# }
```
[the remark this answers is not in the document]
Comment (sanjana) [reply langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal]: Added in https://github.com/ballerina-platform/ballerina-lang/pull/38983/commits/c58f82beaff1f24fca037ef96d7b498087ba5115

File: langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal [hunk 3]
```diff
@@ -40,6 +40,18 @@ public type TypeId readonly & record {|
 |};
 
 # Returns the type-ids induced by a typedesc value.
+# 
+# ```ballerina
+# type Error distinct error;
+# type SampleError distinct (Error & error<record {string msg;}>);
```
Comment (pcnfernando) [inline langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal:46]:
```suggestion
# type Error distinct error;

# type SampleError distinct (Error & error<record {string msg;}>);
```

Comment (sanjana) [reply langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/38983/commits/c857d393985333776f00303de8af5ae5204fa32b

File: langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal [hunk 4]
```diff
@@ -40,6 +40,19 @@ public type TypeId readonly & record {|
 |};
 
 # Returns the type-ids induced by a typedesc value.
+# 
+# ```ballerina
+# type Error distinct error;
+# 
+# type SampleError distinct (Error & error<record {string msg;}>);
+# 
+# typedesc:TypeId[] typeIds = <typedesc:TypeId[]>SampleError.typeIds();
```
Comment (MaryamZi) [inline langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal:49]: Let's not use casts. What I meant in https://github.com/ballerina-platform/ballerina-lang/pull/38983#discussion_r1052029645 was something like

```suggestion
# SampleError.typeIds() ⇒ [{"moduleId":{"organization":"$anon","name":".","platformParts":["0"]},"localId":"SampleError"},{"moduleId":{"organization":"$anon","name":".","platformParts":["0"]},"localId":"Error"}]
```

Comment (mohanvive) [reply langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal:49]: @sanjana please check. And sync with the PR sent to 3.x

Comment (sanjana) [reply langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal]: Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/38983/commits/cde0e8f3a153e92347debe831f799782aaf384ac

**Review by MaryamZi — CHANGES_REQUESTED** [review]
https://github.com/ballerina-platform/ballerina-lang/pull/39111#pullrequestreview-1227101262

File: langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal [hunk 5]
```diff
@@ -40,6 +40,17 @@ public type TypeId readonly & record {|
 |};
 
 # Returns the type-ids induced by a typedesc value.
+# 
+# ```ballerina
+# type Error distinct error;
+# 
+# type SampleError distinct (Error & error<record {string msg;}>);
+#
```
Comment (MaryamZi) [inline langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal:48]:
```suggestion
# 
# Error.typeIds() ⇒ [{"moduleId":{"organization":"$anon","name":".","platformParts":["0"]},"localId":"Error"}]
#
```

Comment (sanjana) [reply langlib/lang.typedesc/src/main/ballerina/typedesc_lib.bal:48]: Updated in https://github.com/ballerina-platform/ballerina-lang/pull/38983/commits/9f0cc0ee0d5845c8997e16410ce2fc154c6bf887

**Review by SasinduDilshara — APPROVED** [review]

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**